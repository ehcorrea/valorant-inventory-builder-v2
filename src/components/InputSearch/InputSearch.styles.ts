import { TextInput, TouchableOpacity } from 'react-native';
import styled, { css } from 'styled-components/native';
import FontAwesome from '@expo/vector-icons/MaterialIcons';

export const Wrapper = styled(TouchableOpacity)`
  ${({ theme }) => css`
    align-items: center;
    background-color: #ffffff29;
    border-radius: 30px;
    flex-direction: row;
    height: ${theme.spacings.vertical.large}px;
    padding-horizontal: ${theme.spacings.rwvalue(8)}px;
    width: 100%;
  `}
`;

export const Icon = styled(FontAwesome).attrs(({ theme }) => ({
  size: theme.spacings.vertical.medium,
}))`
  border-radius: 50px;
`;

export const Input = styled(TextInput).attrs(({ theme }) => ({
  placeholder: 'Search',
  placeholderTextColor: '#ffffff99',
}))`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    flex: 1;
    font-size: ${theme.font.sizes.small}px;
    margin-left: ${theme.spacings.rwvalue(5)}px;
    padding-vertical: ${theme.spacings.rhvalue(8)}px;
  `}
`;
