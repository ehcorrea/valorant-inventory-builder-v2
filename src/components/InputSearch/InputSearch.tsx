import { useRef, useState } from 'react';

import { TextInput, TextInputProps } from 'react-native';

import * as S from './InputSearch.styles';

export type InputSearchProps = TextInputProps;

export function InputSearch({ onChangeText, ...props }: InputSearchProps) {
  const [text, setText] = useState('');
  const inputRef = useRef<TextInput>(null);

  return (
    <S.Wrapper
      onPress={() => {
        inputRef.current?.focus();
      }}
    >
      <S.Icon name="search" />
      <S.Input
        ref={inputRef}
        onChangeText={(value) => {
          setText(value);
          onChangeText?.(value);
        }}
        value={text}
        {...props}
      />
    </S.Wrapper>
  );
}
