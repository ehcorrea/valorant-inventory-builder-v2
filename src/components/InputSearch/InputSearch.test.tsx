import { fireEvent, render } from '@/test/utils';
import { InputSearch } from './InputSearch';

const mockedOnFocus = jest.fn();

const setup = () => {
  return render(<InputSearch onFocus={mockedOnFocus} />);
};

describe('component <InputSearch/> update value', () => {
  test('when change text', () => {
    const container = setup();
    const search = container.getByPlaceholderText('Search');
    expect(search.props.value).toBe('');
    fireEvent.changeText(search, 'lorem ipsum');
    expect(search.props.value).toBe('lorem ipsum');
    expect(container).toMatchSnapshot();
  });
});
