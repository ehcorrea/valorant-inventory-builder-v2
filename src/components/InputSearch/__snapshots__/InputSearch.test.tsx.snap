// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <InputSearch/> update value when change text 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff29",
      "borderBottomLeftRadius": 30,
      "borderBottomRightRadius": 30,
      "borderTopLeftRadius": 30,
      "borderTopRightRadius": 30,
      "flexDirection": "row",
      "height": 51,
      "opacity": 1,
      "paddingHorizontal": 16,
      "width": "100%",
    }
  }
>
  <Text />
  <TextInput
    onChangeText={[Function]}
    onFocus={[MockFunction]}
    placeholder="Search"
    placeholderTextColor="#ffffff99"
    style={
      {
        "color": "#FFFBF5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "fontSize": 27,
        "marginLeft": 10,
        "paddingVertical": 13,
      }
    }
    value="mocked input value"
  />
</View>
`;
